docs/user-guide/creating-application/deployment-template/deployment.md

```diff
@@ -609,6 +609,40 @@ Containers section can be used to run side-car containers along with your main c
           - migrate
 ```
 
+### Container Lifecycle Hooks
+
+Container lifecycle hooks are mechanisms that allow users to define custom actions to be performed at specific stages of a container's lifecycle i.e. PostStart or PreStop.
+
+```yaml
+containerSpec:
+  lifecycle:
+    enabled: false
+    postStart:
+      httpGet:
+        host: example.com
+        path: /example
+        port: 90
+    preStop:
+      exec:
+        command:
+          - sleep
+          - "10"
+```
+
+| Key | Description |
+| :--- | :--- |
+| `containerSpec` | containerSpec to define container lifecycle hooks configuration. |
+| `lifecycle` | Lifecycle hooks for the container. |
+| `enabled` | Set true to enable lifecycle hooks for the container else set false. |
+| `postStart` | The postStart hook is executed immediately after a container is created. |
+| `httpsGet` | Sends an HTTP GET request to a specific endpoint on the Container. |
```

Comment:
| `httpsGet` | Sends an HTTP GET request to a specific endpoint on the container. |